lib/db.js

```diff
@@ -238,6 +238,12 @@ module.exports = (
 
     return P.all(promises)
       .spread((mysqlSessionTokens, redisTokens) => {
+        if (config.tokenLifetimes.sessionTokenWithoutDevice) {
+          // Filter out any expired sessions
+          mysqlSessionTokens = mysqlSessionTokens.filter(sessionToken =>
+            sessionToken.createdAt > Date.now() - config.tokenLifetimes.sessionTokenWithoutDevice
```

Comment:
The alternative here would be to call `SessionToken.fromHex` per item so that we can check using `token.expired`. I decided against that because of the extra work it entails for each item in the array: constructing objects, converting strings to buffers, deriving keys. It seemed worth a small amount of duplication to avoid paying that cost.